modules/compute/gke-node-pool/main.tf

```diff
@@ -104,12 +104,18 @@ resource "google_container_node_pool" "node_pool" {
       }
     }
 
-    ephemeral_storage_local_ssd_config {
-      local_ssd_count = var.local_ssd_count_ephemeral_storage
+    dynamic "ephemeral_storage_local_ssd_config" {
+      for_each = var.local_ssd_count_ephemeral_storage > 0 ? [1] : []
+      content {
+        local_ssd_count = var.local_ssd_count_ephemeral_storage
+      }
     }
 
-    local_nvme_ssd_block_config {
-      local_ssd_count = var.local_ssd_count_nvme_block
+    dynamic "local_nvme_ssd_block_config" {
+      for_each = var.local_ssd_count_nvme_block > 0 ? [1] : []
+      content {
+        local_ssd_count = var.local_ssd_count_nvme_block
+      }
```

Comment:
I don't quite see why these had to be made dynamic blocks? Is there a problem being explicit and assigning zero when the value is zero.

Reply:
Fixed. Now if the value is not null the block will be added.